partiql-tests-data/fail/static-analysis/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,655 @@
+  statement: "TIMESTAMP '2021-10-32 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid DAY field - out of range for November",
+  statement: "TIMESTAMP '2021-11-31 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid DAY field - out of range for December",
+  statement: "TIMESTAMP '2021-12-32 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid DAY field - day of zero",
+  statement: "TIMESTAMP '2021-01-00 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid DAY field - out of range",
+  statement: "TIMESTAMP '2021-03-12345 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid Day field - minus before day",
+  statement: "TIMESTAMP '2021-03--10 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid Day field - plus before day",
+  statement: "TIMESTAMP '2021-03-+10 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid literal Format - random string",
+  statement: "TIMESTAMP 'timestamp'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 24:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - Negative value",
+  statement: "TIMESTAMP '2021-03-01 -01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 +01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 +01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 00:60:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - negative value",
+  statement: "TIMESTAMP '2021-03-01 00:-01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 00:+01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// Setting to 62 to take account of leap second, if any implement does care.
```

Comment:
Just curious does the SQL spec say anything about leap seconds?

Reply:
Yes. Whether or not to support Leap Second is up to the implementation.